packages/ui/src/components/vuestic-components/va-rating/VaRating.vue

```diff
@@ -1,279 +1,206 @@
 import { ColorThemeMixin } from '../../../services/ColorThemePlugin'
-import { ContextPluginMixin, getContextPropValue } from '../../context-test/context-provide/ContextPlugin'
+import { ContextPluginMixin, makeContextablePropsMixin } from '../../context-test/context-provide/ContextPlugin'
+import Component, { mixins } from 'vue-class-component'
+import { Watch } from 'vue-property-decorator'
+import { ColorInput } from 'colortranslator/dist/@types'
+import { SizeMixin } from '../../../mixins/SizeMixin'
+
+const RatingPropsMixin = makeContextablePropsMixin({
+  value: { type: Number, default: 0 },
+  icon: { type: String, default: 'star' },
+  halfIcon: { type: String, default: 'star_half' },
+  emptyIcon: { type: String, default: 'star_empty' },
+  readonly: { type: Boolean, default: false },
+  disabled: { type: Boolean, default: false },
+  numbers: { type: Boolean, default: false },
+  halves: { type: Boolean, default: false },
+  max: { type: Number, default: 5 },
+  size: { type: [String, Number], default: 'medium' },
+  clearable: { type: Boolean, default: false },
+  hover: { type: Boolean, default: false },
+  texts: { type: Array, default: () => [] },
+  textColor: { type: String },
+  unselectedColor: { type: String },
+})
 
-export default {
+@Component({
   name: 'VaRating',
   components: { VaRatingItem },
-  mixins: [ColorThemeMixin, ContextPluginMixin],
-  props: {
-    value: {
-      type: Number,
-      default () {
-        return getContextPropValue(this, 'value', 0)
-      },
-    },
-    icon: {
-      type: String,
-      default () {
-        return getContextPropValue(this, 'icon', 'fa fa-star')
-      },
-    },
-    halfIcon: {
-      type: String,
-      default () {
-        return getContextPropValue(this, 'icon', 'fa fa-star-half-full')
-      },
-    },
-    emptyIcon: {
-      type: String,
-      default () {
-        return getContextPropValue(this, 'emptyIcon', '')
-      },
-    },
-    readonly: {
-      type: Boolean,
-      default () {
-        return getContextPropValue(this, 'readOnly', false)
-      },
-    },
-    disabled: {
-      type: Boolean,
-      default () {
-        return getContextPropValue(this, 'disabled', false)
-      },
-    },
-    numbers: {
-      type: Boolean,
-      default () {
-        return getContextPropValue(this, 'numbers', false)
-      },
-    },
-    halves: {
-      type: Boolean,
-      default () {
-        return getContextPropValue(this, 'halves', false)
-      },
-    },
-    max: {
-      type: Number,
-      default () {
-        return getContextPropValue(this, 'max', 5)
-      },
-    },
-    size: {
-      type: String,
-      default () {
-        return getContextPropValue(this, 'size', 'medium')
-      },
-    },
-  },
-  data () {
+})
+export default class VaRating extends mixins(
+  RatingPropsMixin,
+  ColorThemeMixin,
+  ContextPluginMixin,
+  SizeMixin,
+) {
+  private readonly HALF_VALUE = 0.5
```

Comment:
You can move types to separate file?
I see you used some for rating item as well?